editor/scene_tree_dock.cpp

```diff
@@ -76,6 +76,41 @@ void SceneTreeDock::_quick_open() {
 	instantiate_scenes(quick_open->get_selected_files(), scene_tree->get_selected());
 }
 
+void SceneTreeDock::_handle_hover_to_inspect() {
+	if (!tree_clicked) {
+		scene_tree->set_marked(nullptr, false, false);
+		return;
+	}
+
+	Tree *tree = scene_tree->get_scene_tree();
+	TreeItem *item = tree->get_item_at_position(tree->get_local_mouse_position());
+	if (item) {
+		NodePath np = item->get_metadata(0);
+		node_hovered_now = get_node_or_null(np);
+		if (node_hovered_previously != node_hovered_now) {
+			inspect_hovered_node_delay->start();
+		}
+		node_hovered_previously = node_hovered_now;
+	} else {
+		_reset_hovering_timer();
+	}
+}
+
+void SceneTreeDock::_inspect_hovered_node() {
+	if (node_hovered_previously == node_hovered_now) {
```

Comment:
When hovered node changes the timer is reset, so the condition is unnecessary, no?